### PR TITLE
Remove CachingDirectoryManager (and proper-lockfile usage)

### PR DESCRIPTION
**Pull request by ipcrm**
Fixes https://github.com/atomist/automation-client/issues/225.

### CODE REVIEW

File: lib/spi/clone/CachingDirectoryManager.ts [hunk 1]
```diff
@@ -110,26 +110,26 @@ type LockResult = LockAcquired | NoLockForYou;
 
 function pleaseLock(lockPath: string): Promise<LockResult> {
     return new Promise<LockResult>((resolve, reject) => {
```
Comment (ddgenome) [inline lib/spi/clone/CachingDirectoryManager.ts]: It looks like the new version of lockfile.lock supports returning a promise, so wrapping in a promise seems unnecessary and this method could be greatly simplified.

File: test/project/git/LockfileLibrary.test.ts [hunk 2]
```diff
@@ -7,14 +7,19 @@ import {
 
 describe("exploratory tests for proper-locking", () => {
```
Comment (ddgenome) [inline test/project/git/LockfileLibrary.test.ts]: This test seems to just test lockfile, not our code.

Comment (ipcrm) [reply test/project/git/LockfileLibrary.test.ts]: Indeed it does.  I don't believe it has any value - but I wasn't sure if it was put there to be a "canary in the mine" as it turned out to be in the failure from #225.  Should I just remove this?

File: lib/spi/clone/CachingDirectoryManager.ts (hunk 1 above)
Comment (ipcrm) [reply lib/spi/clone/CachingDirectoryManager.ts]: I'll take a crack at revising this, or removing, as mentioned https://github.com/atomist/automation-client/pull/639#pullrequestreview-336622281

**Comment by ipcrm** [comment]
Updated to:
* Removed `pleaseLock` abstraction of `proper-lockfile`
* Remove unneeded tests
* Simplify logic using async/await
* Updated "clones a project subdirectory" test timeout to 15 seconds after having inconsistent executions that were taking 10-11 seconds and exceeding the existing timeout

File: lib/spi/clone/CachingDirectoryManager.ts [hunk 3]
```diff
@@ -36,38 +37,37 @@ const cache = new StableDirectoryManager({
...
+                release: () => {
+                    logger.debug("Releasing lock on '%s'", existing.path);
+                    return lockResult().then(existing.release);
+                },
+                invalidate: () => {
+                    logger.debug("Invalidating '%s'", existing.path);
+                    return cache.invalidate(existing)
+                        .then(() => {
+                            logger.debug("Invalidated. Now releasing lock");
+                            return lockResult().then(existing.release);
```
Comment (ipcrm) [inline lib/spi/clone/CachingDirectoryManager.ts]: I'm going to investigate `existing.release` further.  I"m not sure how this is actually consumed.

**Comment by ipcrm** [comment]
@ddgenome @cdupuis 
Having dug into this further, I cannot find a place where this directory manager is even being used.  The repo loaders (https://github.com/atomist/automation-client/search?q=DefaultDirectoryManager&unscoped_q=DefaultDirectoryManager) are all making use of `DefaultDirectoryManager` which is defined here: https://github.com/atomist/automation-client/blob/3ea459ab3e1b171e1b3d02bd5ebb67c599284804/lib/project/git/GitCommandGitProject.ts#L53.  It's not clear to me how CachingDirectoryManager directory manager would get utilized unless the expectation is that it's only for custom repoLoaders.

In light of this, I'm proposing we simply remove it, along with the `proper-lockfile` package as this is the only place it's used.  That said, I'm not sure if there some usage I'm not aware of so looking for feedback.

**Comment by ipcrm** [comment]
Having dug into this further, I cannot find a place where this directory manager is being used (outside tests).  The repo loaders (https://github.com/atomist/automation-client/search?q=DefaultDirectoryManager&unscoped_q=DefaultDirectoryManager) are all making use of `DefaultDirectoryManager` which is defined here: https://github.com/atomist/automation-client/blob/3ea459ab3e1b171e1b3d02bd5ebb67c599284804/lib/project/git/GitCommandGitProject.ts#L53  It's not clear to me how `CachingDirectoryManager` directory manager would get utilized unless the expectation is that it's only for custom repoLoaders.

In light of this, I'm proposing we simply remove it, along with the `proper-lockfile` package as this is the only place it's used.  That said, I'm not sure if there some usage I'm not aware of so looking for feedback.

File: package.json [hunk 4]
```diff
@@ -132,6 +132,7 @@
   },
   "devDependencies": {
     "@types/mocha": "^5.2.7",
+    "@types/proper-lockfile": "^4.1.1",
```
Comment (ddgenome) [inline package.json]: This should be a regular dependency since automation-client is a dependency to other packages.

**Comment by ddgenome** [comment]
It seems the behavior I am thinking about is provided by the CachingProjectLoader in sdm.

**Comment by ipcrm** [comment]
Yea, as you mention with the CachingProjectLoader, it seems like the original intent of this directory manager ultimately got implemented elsewhere.

File: package.json (hunk 4 above)
Comment (ipcrm) [reply package.json]: Updated.

**Comment by cdupuis** [comment]
👍 for removing it.

**Pull request merged**